src/Widgets/PowerModeList.vala

```diff
@@ -0,0 +1,174 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ * SPDX-FileCopyrightText: 2024 elementary, Inc. (https://elementary.io)
+ */
+
+
+public class Power.Widgets.PowerModeList : Gtk.Box {
+    public static bool successfully_initialized { get; private set; default = true; }
+
+    private static Services.DBusInterfaces.PowerProfile? pprofile;
+    private static GLib.Settings? settings;
+
+    public string settings_key { get; construct; }
+
+    private Gtk.RadioButton saver_radio;
+    private Gtk.RadioButton balanced_radio;
+    private Gtk.RadioButton performance_radio;
+
+    private const string ICON_RES = "/io/elementary/desktop/wingpanel/power/scalable/categories/";
+
+    public PowerModeList (bool on_battery) {
+        Object (
+            settings_key: on_battery ? "profile-on-good-battery" : "profile-plugged-in",
+            orientation: Gtk.Orientation.VERTICAL,
+            margin: 6
+        );
+    }
+
+    static construct {
+        var schema = SettingsSchemaSource.get_default ().lookup ("io.elementary.settings-daemon.power", true);
+        if (schema != null && schema.has_key ("profile-plugged-in") && schema.has_key ("profile-on-good-battery")) {
+            settings = new GLib.Settings ("io.elementary.settings-daemon.power");
+        } else {
+            warning ("settings-daemon schema not found, will connect to power-profiles-daemon directly");
+        }
+
+        try {
+            pprofile = Bus.get_proxy_sync (
+                BusType.SYSTEM,
+                Services.DBusInterfaces.POWER_PROFILES_DAEMON_NAME,
+                Services.DBusInterfaces.POWER_PROFILES_DAEMON_PATH,
+                DBusProxyFlags.NONE
+            );
+        } catch (Error e) {
+            critical (e.message);
+            successfully_initialized = false;
+        }
+    }
+
+    construct {
+        if (pprofile == null) {
+            return;
+        }
+
+        build_ui ();
+        build_events ();
+    }
+
+    private void build_ui () {
+        Gtk.IconTheme.get_default ().add_resource_path ("/io/elementary/desktop/wingpanel/power");
```

Comment:
I would really prefer this to be in the `Indicator` class since we probably will resource the battery icons for the display widget in the future so we don't want to duplicate adding the resource path

Reply:
Done